doc/contributing/submitting-changes.chapter.md

```diff
@@ -83,6 +83,62 @@ If a security fix applies to both master and a stable release then, similar to r
 
 Critical security fixes may by-pass the staging branches and be delivered directly to release branches such as `master` and `release-*`.
 
+## Writing package tests {#submitting-changes-package-tests}
+
+Tests are important to improve quality and make reviews and automatic updates easy.
+
+Nix package tests are a lightweight alternative to NixOS module tests. They can be used to create simple integration tests for packages while the module tests are used to test services or programs with a graphical user interface on a NixOS VM.
+
+This is an example using the `phoronix-test-suite` package with the current best practice.
+
+Add this to the package definition:
+
+```
+passthru.tests = {
+  simple-execution = callPackage ./tests.nix { };
+};
+```
+
+Create `tests.nix` in the package directory:
+
+```
+{ runCommand, phoronix-test-suite }:
+
+let
+  inherit (phoronix-test-suite) pname version;
+in
+
+runCommand "${pname}-tests" { meta.timeout = 3; }
+  ''
+    # automatic initial setup to prevent interactive questions
+    ${phoronix-test-suite}/bin/phoronix-test-suite enterprise-setup >/dev/null
+    # get version of installed program and compare with package version
+    if [[ `${phoronix-test-suite}/bin/phoronix-test-suite version` != *"${version}"*  ]]; then
+      echo "Error: program version does not match package version"
+      exit 1
+    fi
+    # run dummy command
+    ${phoronix-test-suite}/bin/phoronix-test-suite dummy_module.dummy-command >/dev/null
+    # needed for Nix to register the command as successful
+    touch $out
+  ''
+```
+
+You can run these tests with:
+
+```
```

Comment:
```suggestion
```ShellSession
```

Reply:
where do you got that from? github uses `shell` for example

https://stackoverflow.com/a/20304739

Reply:
https://github.com/github/linguist/blob/32ec19c013a7f81ffaeead25e6e8f9668c7ed574/lib/linguist/languages.yml#L5415